### ISSUE TITLE
High CPU usage for stdclient and payload

### ISSUE THREAD

**Issue by nikkout** [fortio/fortio#505]
When using the fortio with the stdclient option and payload (either with payload-size or payload and a string) the CPU usage reaches more than 300% which is way more than the fast client is using (< 10% ) for the same traffic configuration.
For the traffic I am using 1000qps and 8 connections against a fortio server.
The fortio version is 1.20.0 2021-12-24 21:20 4a325b93146c7faf5d7ff5a327d3c86abceb141f go1.17.3

**Comment by ldemailly** [comment]
are you using the uuid or just plain requests? how big a payload?

**Comment by nikkout** [comment]
I am using plain requests and I have tried with payloads of size 1024 and also just 1. The full command I used look like this:
"fortio load -qps 1000 -stdclient -c 8 -t 180s -payload-size 1024"

**Comment by nikkout** [comment]
Any update on this issue?

**Comment by ldemailly** [comment]
hey sorry I missed you update from 2 weeks ago (somehow github doesn't tell me about all updates)

mid term I would like to support TLS in the fast client too but let me see if there is something we can do with the go std client

which go version ? did you try a few? did you try profiling ?

**Comment by nikkout** [comment]
Hello, I have not used profiling and I used the latest docker image provided for fortio.

**Comment by ldemailly** [comment]
@nikkout So I tried and I can't reproduce such a big difference, can you show the full output of your test with  time in front and which OS are you on

For me on MacOS x86:
Standard client:
```
$ time fortio load -qps 1000 -stdclient -c 8 -t 180s -payload-size 1024 http://localhost:8080
Fortio 1.22.0 running at 1000 queries per second, 16->16 procs, for 3m0s: http://localhost:8080
17:26:02 I httprunner.go:87> Starting http test for http://localhost:8080 with 8 threads at 1000.0 qps and parallel warmup
Starting at 1000 qps with 8 thread(s) [gomax 16] for 3m0s : 22500 calls each (total 180000)
17:29:02 I periodic.go:693> T000 ended after 3m0.000772975s : 22500 calls. qps=124.99946321411069
17:29:02 I periodic.go:693> T002 ended after 3m0.000775717s : 22500 calls. qps=124.99946130996038
17:29:02 I periodic.go:693> T004 ended after 3m0.000852626s : 22500 calls. qps=124.99940790141576
17:29:02 I periodic.go:693> T001 ended after 3m0.000886773s : 22500 calls. qps=124.99938418845048
17:29:02 I periodic.go:693> T007 ended after 3m0.000889429s : 22500 calls. qps=124.99938234402423
17:29:02 I periodic.go:693> T005 ended after 3m0.00089374s : 22500 calls. qps=124.99937935030388
17:29:02 I periodic.go:693> T003 ended after 3m0.000929181s : 22500 calls. qps=124.99935473874758
17:29:02 I periodic.go:693> T006 ended after 3m0.00095598s : 22500 calls. qps=124.99933612852583
Ended after 3m0.000990923s : 180000 calls. qps=999.99
Sleep times : count 179992 avg 0.0066878008 +/- 0.001609 min -0.095656958 max 0.007862025 sum 1203.75065
Aggregated Function Time : count 180000 avg 0.00054805133 +/- 0.0007124 min 0.000107942 max 0.103435151 sum 98.6492388
# range, mid point, percentile, count
>= 0.000107942 <= 0.001 , 0.000553971 , 98.15, 176677
> 0.001 <= 0.002 , 0.0015 , 99.96, 3257
> 0.002 <= 0.003 , 0.0025 , 99.99, 43
> 0.003 <= 0.004 , 0.0035 , 99.99, 13
> 0.005 <= 0.006 , 0.0055 , 100.00, 2
> 0.1 <= 0.103435 , 0.101718 , 100.00, 8
# target 50% 0.000562358
# target 75% 0.000789568
# target 90% 0.000925894
# target 99% 0.00146761
# target 99.9% 0.001965
Uniform: false, Jitter: false
Code 200 : 180000 (100.0 %)
Response Header Sizes : count 180000 avg 0 +/- 0 min 0 max 0 sum 0
Response Body/Total Sizes : count 180000 avg 1024 +/- 0 min 1024 max 1024 sum 184320000
All done 180000 calls (plus 8 warmup) 0.548 ms avg, 1000.0 qps

real	3m0.025s
user	0m35.523s
sys	0m34.703s
```
total of 35.5+34.7 = 70.2 seconds CPU

Fast client:
```
$ time fortio load -qps 1000 -c 8 -t 180s -payload-size 1024 http://localhost:8080
Fortio 1.22.0 running at 1000 queries per second, 16->16 procs, for 3m0s: http://localhost:8080
17:29:25 I httprunner.go:87> Starting http test for http://localhost:8080 with 8 threads at 1000.0 qps and parallel warmup
Starting at 1000 qps with 8 thread(s) [gomax 16] for 3m0s : 22500 calls each (total 180000)
17:32:25 I periodic.go:693> T003 ended after 3m0.001177375s : 22500 calls. qps=124.99918238382023
17:32:25 I periodic.go:693> T007 ended after 3m0.001344924s : 22500 calls. qps=124.99906603197843
17:32:25 I periodic.go:693> T001 ended after 3m0.001361078s : 22500 calls. qps=124.99905481409151
17:32:25 I periodic.go:693> T004 ended after 3m0.001379531s : 22500 calls. qps=124.99904199970328
17:32:25 I periodic.go:693> T006 ended after 3m0.001361047s : 22500 calls. qps=124.99905483561896
17:32:25 I periodic.go:693> T002 ended after 3m0.001383878s : 22500 calls. qps=124.99903898099961
17:32:25 I periodic.go:693> T005 ended after 3m0.00140681s : 22500 calls. qps=124.99902305624653
17:32:25 I periodic.go:693> T000 ended after 3m0.001410722s : 22500 calls. qps=124.99902033962238
Ended after 3m0.001494905s : 180000 calls. qps=999.99
Sleep times : count 179992 avg 0.006342973 +/- 0.0006482 min -0.000544767 max 0.007871212 sum 1141.6844
Aggregated Function Time : count 180000 avg 0.00044539221 +/- 0.0001576 min 7.5302e-05 max 0.007697612 sum 80.170598
# range, mid point, percentile, count
>= 7.5302e-05 <= 0.001 , 0.000537651 , 99.47, 179044
> 0.001 <= 0.002 , 0.0015 , 99.99, 945
> 0.005 <= 0.006 , 0.0055 , 99.99, 1
> 0.006 <= 0.007 , 0.0065 , 100.00, 2
> 0.007 <= 0.00769761 , 0.00734881 , 100.00, 8
# target 50% 0.000540117
# target 75% 0.000772527
# target 90% 0.000911973
# target 99% 0.000995641
# target 99.9% 0.00182116
Sockets used: 8 (for perfect keepalive, would be 8)
Uniform: false, Jitter: false
Code 200 : 180000 (100.0 %)
Response Header Sizes : count 180000 avg 118 +/- 0 min 118 max 118 sum 21240000
Response Body/Total Sizes : count 180000 avg 1142 +/- 0 min 1142 max 1142 sum 205560000
All done 180000 calls (plus 8 warmup) 0.445 ms avg, 1000.0 qps

real	3m0.023s
user	0m14.025s
sys	0m28.403s
```
Total CPU used 14.0+28.4 = 42.4 s

So the std client for me is 1.65x less efficient not 30x like your description

**Comment by ldemailly** [comment]
I'm able to reproduce with 1.20

15:01:08 E http_client.go:376> [7] Unable to send POST request for http://localhost:8080 : Post "http://localhost:8080": dial tcp [::1]:8080: connect: resource temporarily unavailable

This is actually a side effect of the bug that was only fixed in 1.21.2 #519

@nikkout I should have looked sooner and could have been fixed when you first reported it - but fixed now!

Thanks
